### DISCUSSION TITLE
kklib/src/all.c missing in stack build?

### DISCUSSION THREAD

**Discussion by osa1** [koka-lang/koka#648] [Q&A]
When I build koka with `stack build` and call the generated executable with the right `-i` and `--cclibdir` arguments, I still get this error when compiling to C:

```
cc1: fatal error: /home/omer/koka/koka/.stack-work/dist/x86_64-linux-tinfo6/ghc-9.6.4/build/koka/share/koka/v3.1.3/kklib/src/all.c: No such file or directory
compilation terminated.
test4/@main(1, 1): build error: user error (error  : command failed (exit code 1)
command: /usr/bin/gcc -Wall -Wextra -Wpointer-arith -Wshadow -Wstrict-aliasing -Wno-unknown-pragmas -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-variable -Wno-unused-value -Wno-unused-but-set-variable -g -Og -c -I /home/omer/koka/koka/.stack-work/dist/x86_64-linux-tinfo6/ghc-9.6.4/build/koka/share/koka/v3.1.3/kklib/include -I /home/omer/koka/koka/kklib/include -I /home/omer/koka/koka/.stack-work/dist/x86_64-linux-tinfo6/ghc-9.6.4/build/koka/share/koka/v3.1.3/kklib/mimalloc/include -DKK_MIMALLOC=8 "-DKK_COMP_VERSION=\"3.1.3\"" "-DKK_CC_NAME=\"gcc\"" -o .koka/v3.1.3/gcc-debug-33c8ed/kklib.o /home/omer/koka/koka/.stack-work/dist/x86_64-linux-tinfo6/ghc-9.6.4/build/koka/share/koka/v3.1.3/kklib/src/all.c)
```

To work around this I copy kklib directory to `.stack-work/dist/x86_64-linux-tinfo6/ghc-9.6.4/build/koka/share/koka/v3.1.3/kklib`, but I wonder if there's a flag that I need to be passing?

**Comment by TimWhiting** [comment]
I've typically only used the stack built executable after bundling (see util/bundle.kk). 

If you have local changes to the compiler / are on the dev branch there are several easier ways to deal with this though.

To build+run you can use `stack run koka -- ...args...`. (You'll have to be in the koka repo directory, and specify relative / absolute paths to your project). 
To build you can use `stack build`, or `stack build --fast`. 
To then run a previous build you can use `stack exec koka -- ...args...`.

However, the easiest way to work with the local repo checkout is to use vscode and add the `koka.dev.developmentPath` setting to an absolute path to where you cloned the repo (including `koka`). Then restart the extension / reload your window. This will essentially run `stack exec koka` from within your developmentPath directory, and supply the absolute path to your project and add the project path to the `-i` directories as well. 

Also, if you are playing around with JS, there is an update to the extension in `support/vscode/koka.language-koka` for better js support. For example I can use the following in my launch config:
```koka
{
      "type": "koka",
      "request": "launch",
      "program": "test/jsgen/err",
      "name": "test/jsgen/err",
      "programArgs": ["test/jsgen/err.kk"]
 },
```
Then I can set the backend to javascript using: `Cmd/Ctrl+P` -> Koka: Set compilation target. Set it to nodejs. Then you should be able to run the launch config to run the program. If you don't need program arguments, you can just use the code action hint.

To build the extension, use `npm install && npm run build`. Then to install right click the `.vsix` that is produced and install it.
